eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml

```diff
@@ -12,21 +12,41 @@ parameters:
   default: []
 - name: JobTemplatePath
   type: string
+# Set this to false to do a full checkout for private repositories with the azure pipelines service connection
+- name: SparseCheckout
+  type: boolean
+  default: true
 
 jobs:
 - job: generate_matrix
   variables:
     displayNameFilter: $[ coalesce(variables.jobMatrixFilter, '.*') ]
   pool:
-    name: Azure Pipelines
-    vmImage: ubuntu-18.04
+    name: azsdk-pool-mms-ubuntu-1804-general
+    vmImage: MMSUbuntu18.04
   displayName: Generate Job Matrix
   steps:
+    - ${{ if parameters.SparseCheckout }}:
+      - checkout: none
+
+      - pwsh: |
+          git clone --no-checkout --filter=tree:0 git://github.com/$(Build.Repository.Name).git .
```

Comment:
You'll also need to consider auth for the ```-pr``` repositories.

Reply:
I think the initial approach @weshaggard and I were thinking about is to just do the full checkout for private, since it includes the service connection, but I'll need to find the right place to set `SparseCheckout: false` for that.